src/modules/poweraccent/PowerAccent/Program.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.CommandLine;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using interop;
+using ManagedCommon;
+using PowerAccent.UI;
+
+namespace PowerAccent;
+
+static class Program
+{
+    private const string PROGRAM_NAME = "PowerAccent";
+    private static App _application;
+    private static Mutex _mutex = new Mutex(true, "PowerToys.PowerAccent");
+
+    [STAThread]
+    static void Main(string[] args)
+    {
+        Arguments(args);
+
+        if (_mutex.WaitOne(TimeSpan.Zero, true))
+        {
+            InitEvents();
+
+            _application = new App();
+            _application.InitializeComponent();
+            //(_application.MainWindow as Selector).HideTaskbarIcon();
+            _application.Run();
+        }
+    }
+
+    private static void InitEvents()
+    {
+        // Detect exit event from PowerToys
+        new Thread(() =>
+        {
+            EventWaitHandle eventHandle = new EventWaitHandle(false, EventResetMode.AutoReset, Constants.PowerAccentExitEvent());
+            if (eventHandle.WaitOne())
+            {
+                Terminate();
+            }
+        }).Start();
+
+        // wait for 
+    }
+
+    #region Arguments
+
+    private static int Arguments(string[] args)
+    {
+        Option<int> pidOption = new(
```

Comment:
@damienleroy did you base your program.cs off Awake?

Reply:
@jaimecbernardo this may be able to be cleaned up a bit if this was based on Awake

Reply:
Yes exactly. Is there an another way to do it so? (better way?)